src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -822,6 +849,30 @@ theorem sshiftRight_eq_of_msb_true {x : BitVec w} {s : Nat} (h : x.msb = true) :
         Nat.not_lt, decide_eq_true_eq]
       omega
 
+theorem sshiftRight_xor_distrib (x y : BitVec w) (n : Nat) :
+    (x ^^^ y).sshiftRight n = (x.sshiftRight n) ^^^ (y.sshiftRight n) := by
+  ext i
+  simp
+  split <;>
+  by_cases w ≤ i <;>
+  simp [*]
+
+theorem sshiftRight_and_distrib (x y : BitVec w) (n : Nat) :
+    (x &&& y).sshiftRight n = (x.sshiftRight n) &&& (y.sshiftRight n) := by
+  ext i
+  simp
+  split <;>
+  by_cases w ≤ i <;>
+  simp [*]
```

Comment:
```suggestion
  split 
    <;> by_cases w ≤ i 
    <;> simp [*]
```

I'm not sure what the style-guide is for the "<;>" operator, but at least in 
https://github.com/leanprover/lean4/blob/7776852d227d4269e34fe30eb51ba0adef90838e/src/Init/Data/BitVec/Lemmas.lean#L683
it appears that the new line is indented when "<;>" is used.


The same idea applies on lines 871-874. This is up to you, as I haven't seen this rule documented officially.

Reply:
Right. I now follow this style. Thank you.